src/de/arg/value_parser.rs

```diff
@@ -0,0 +1,47 @@
+use clap::builder::ValueParser;
+use serde::Deserialize;
+
+enum_de!(ValueParser, ValueParser1,
+    #[derive(Deserialize, Clone, Copy)]
+    #[cfg_attr(feature = "kebab-case-key" ,serde(rename_all = "kebab-case"))]
+    #[cfg_attr(feature = "snake-case-key" ,serde(rename_all = "snake_case"))]
+    { }
+    {
+        Bool => {
+            ValueParser::bool()
+        },
+        String => {
+            ValueParser::string()
+        },
+        OsString => {
+            ValueParser::os_string()
+        },
+        PathBuf => {
+            ValueParser::path_buf()
+        },
+        Boolish => {
+            clap::builder::BoolishValueParser::new().into()
+        },
+        Falsey => {
+            clap::builder::FalseyValueParser::new().into()
+        },
+        NonEmptyString => {
+            clap::builder::NonEmptyStringValueParser::new().into()
+        },
+        RangedI64(s : i64 , e : i64) => {
+            (s..e).into()
+        },
+        RangedInclusizeI64(s : i64 , e : i64) => {
+            (s..=e).into()
+        },
+        RangedFromI64(s : i64) => {
+            (s..).into()
+        },
+        RangedToI64(e : i64) => {
+            (..e).into()
+        },
+        RangedToInclusizeI64(e : i64) => {
+            (..=e).into()
+        },
```

Comment:
Would it work to only have one range type and rely on the keys under it to create the range?

Reply:
I used this [`From`](https://docs.rs/clap/3.2.16/clap/builder/struct.ValueParser.html#impl-From%3CRangeTo%3Ci64%3E%3E-for-ValueParser) impl as a reference.

Reply:
The `From` trait exists just for convenience creation and we are limited on what we can handle due to lack of specialization.  There are many more options support for creating value parsers within clap without going down the rabbit hole of custom impls.

Reply:
I'd recommend using `clap::value_parser!` as a guide; that is what `clap_derive` uses.

Reply:
Implemented at #43